src/industry_gui.cpp

```diff
@@ -1449,11 +1485,28 @@ class IndustryDirectoryWindow : public Window {
 			cargos.emplace_back(i->produced_cargo[j], i->last_month_production[j], cargo_suffix[j].text, ToPercent8(i->last_month_pct_transported[j]));
 		}
 
-		/* Sort by descending production, then descending transported */
-		std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
-			if (std::get<1>(a) != std::get<1>(b)) return std::get<1>(a) > std::get<1>(b);
-			return std::get<3>(a) > std::get<3>(b);
-		});
+		switch (this->industries.SortType()) {
+			case IDW_SORT_BY_NAME:
+			case IDW_SORT_BY_TYPE:
+			case IDW_SORT_BY_PRODUCTION:
+				/* Sort by descending production, then descending transported */
+				std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
+					if (std::get<1>(a) != std::get<1>(b)) return std::get<1>(a) > std::get<1>(b);
+					return std::get<3>(a) > std::get<3>(b);
+				});
+				break;
+
+			case IDW_SORT_BY_TRANSPORTED:
+				/* Sort by descending transported, then descending production */
+				std::sort(cargos.begin(), cargos.end(), [](const CargoInfo a, const CargoInfo b) {
+					if (std::get<3>(a) != std::get<3>(b)) return std::get<3>(a) > std::get<3>(b);
+					return std::get<1>(a) > std::get<1>(b);
+				});
```

Comment:
Probably best to turn CargoInfo into a proper struct, so these rather ugly sort functions get better names

Reply:
I'm not sure how to do this

Reply:
Instead of 
```
typedef std::tuple<CargoID, uint16, const char*, uint> CargoInfo;
```
use
```
struct CargoInfo {
    Foo foo;
    Bar bar;
    Baz baz;
};
```

Then instead of `std::get<3>(a)` it becomes `a.baz` etc
(You can define the struct within the function, that's fine)